### PR TITLE
(#3488) NCIDS Hero

### LINKED ISSUE #3487: Story: Build NCIDS Call to Action Strip
## As a content editor, I want be able to add a NCIDS Call to Action Strip component, so that I can embed a NCIDS Call to Action Strip module onto Cancer.gov content.

This story is to implement the NCIDS Call to Action Strip paragraph for use on the Home & Landing Page content type.

### Acceptance Criteria for Analytics
- Component: CTA Strip
- Button text
- Type of link: internal, external, media
- index of button in list of buttons
- number of buttons
- index of component on page
- number of row components on page
- location ("body" maybe)
- current page - the EDDL extension in launch captures and passes this value for all click events

> **ESTIMATE** 5

### Acceptance Criteria
```gherkin
Scenario: Content editor is able to add a "NCIDS Call to Action Strip" to a Home and Landing page
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to Contents
Then the dropdown displays the "Add NCIDS Call to Action Strip" option
When the user selects the "Add NCIDS Call to Action Strip" option
Then the fields for a "NCIDS Call to Action Strip" paragraph appear
  And the Links field appears
  And the field is required
  And it is a paragraph with the options "Add NCIDS Link Button Internal" and “Add NCIDS Link Button External”
  And it shows that it is empty by displaying text that reads "No Links added yet."
  And the dropdown button defaults to "Add NCIDS Link Button Internal"
```

```gherkin
Scenario: Content editor is able to add Link Buttons to a "NCIDS Call to Action Strip"
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user has added a "Add NCIDS Call to Action Strip" to the "Contents" paragraph
When the user selects "Add NCIDS Link Button Internal" for the "Links" paragraph
Then the fields for an "NCIDS Link Button Internal" paragraph appear
  And the first field will be for the "Featured Item"
  And the Featured Item field will be required
  And the Featured Item field will be an entity reference to a node
  And the second field will be for the "Button Text"
  And the field will be required
  And the field will be a text input
  And the third field will be for the "Button Alt Text"
  And the field will be optional
  And the field will be a text input
When the user selects "Add NCIDS Link Button External" for the "Links" paragraph
Then the fields for an "NCIDS Link Button External" paragraph appear
  And the "Featured Item URL" field appears
  And the field is required
  And it is a text input
  And the second field will be for the "Button Text"
  And the field will be required
  And the field will be a text input
  And the third field will be for the "Button Alt Text"
  And the field will be optional
  And the field will be a text input
 ```

 ```gherkin
Scenario: Content editor adds a "NCIDS Call to Action Strip" to a Home and Landing page
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to "Contents"
  And the user selects the "Add NCIDS Call to Action Strip" option
Then the new "NCIDS Call to Action Strip" paragraph appears
When the user adds three links to the Links field
Then the user cannot add any more links to the Links field
  And the user can save the content item
When the user views the Home & Landing Page on cancer.gov
Then the "NCIDS Call to Action Strip" displays
 And the Call to Action Strip section displays centered on the page and in the dark theme
 And the calls to action buttons display
```

### Resources:
Component variant design: [NCIDS #548](https://github.com/NCIOCPL/ncids/issues/548)

### Notes

---

### Prerequisites
- #3551 
- #3519 
- #3744 
- #3745 

### Tasks
- [ ] Create NCIDS Call to Action Strip paragraph
- [ ] Create twig template with NCIDS styling

### Technical Notes
Content Modeling:
- [NCIDS Call to Action Strip paragraph](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Paragraph-Type-NCIDS-Call-To-Action-Strip)
- [NCIDS Link Button](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-link-button)

### LINKED ISSUE #3488: Story: Build NCIDS Hero
## As a content editor, I want be able to add an NCIDS Hero component, so that I can embed a NCIDS Hero module onto Cancer.gov content.

This story is to implement the NCIDS Hero paragraph for use on the Home & Landing Page content type. For MVP2, the prioritized variants are:
- Hero with Tagline - Light Theme
- Hero with Primary Call to Action - Light Theme
- Hero with Primary and CTA Strip - Light Theme

### Acceptance Criteria for Analytics
- Component: hero
- Variant: with light primary CTA; with light primary CTA and CTA strip
  - Other variants:
    - light or dark tagline (default)
    - light tagline with CTA strip
    - dark primary CTA
- Visual viarant
- Functional variant
- Type of link: internal, external
- Button text
- index of component on page
- number of row components on page
- location ("body" maybe)
- current page - the EDDL extension in launch captures and passes this value for all click events

> **ESTIMATE** 5

### Behaviors
- Theme: changes Primary Call to Action section background color
- Image: further questions about image crops/etc
- Tagline: displays in Primary Call to Action section
- Primary Call to Action link
  - Do not draw button if there is no item in the field
  - Do not draw button if there is an unpublished Internal Link item in the field
- Secondary Calls to Action Links
  - Do not draw section unless there are links in this field
  - Do not draw the button for an unpublished Internal Link item in the field

### Acceptance Criteria
```gherkin
Scenario: Content editor is able to add an "NCIDS Hero" to a Home and Landing page
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to Contents
Then the dropdown displays the "Add NCIDS Hero" option
When the user selects the "Add NCIDS Hero" option
Then the fields for a "NCIDS Hero" paragraph appear
  And the "Theme" field appears
  And the field is required
  And it is a dropdown select
  And it has the options "Light Theme" and "Dark Theme"
  And the "Widescreen Image" field appears
  And the field is required
  And it is an image upload
  And the "Desktop Image" field appears
  And the field is required
  And it is an image upload
  And the "Tablet Large Image" field appears
  And the field is required
  And it is an image upload
  And the "Tablet Image" field appears
  And the field is required
  And it is an image upload
  And the "Mobile Large Image" field appears
  And the field is required
  And it is an image upload
  And the "Mobile Image" field appears
  And the field is required
  And it is an image upload
  And the "Tagline" field appears
  And the field is required
  And it is a text input
  And the "Primary Call to Action Link" field appears
  And the field is optional
  And it is an existing paragraph
  And it has the options "Add NCIDS Link Button Internal" and "Add NCIDS Link Button External"
  And it defaults to "Add NCIDS Link Button Internal"
  And the "Secondary Calls to Action Links" field appears
  And the field is optional
  And it has the options "Add NCIDS Call to Action Strip"
```

```gherkin
Scenario: Content editor is able to add a link to the "Primary Call to Action" field on the "NCIDS Hero" 
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user has added a "NCIDS Hero" to the "Contents" paragraph
When the user selects "NCIDS Link Button Internal" for the "Primary Call to Action" paragraph
Then the fields for an "NCIDS Link Button Internal" paragraph appear
  And the "Featured Item" field appears
  And the Featured Item field will be required
  And the Featured Item field will be an entity reference to a node
  And the "Button Text" field appears
  And the field will be required
  And the field will be a text input
  And the "Button Alt Text" field appears
  And the field will be optional
  And the field will be a text input
When the user selects "NCIDS Link Button External" for the "Primary Call to Action" paragraph
Then the fields for an "NCIDS Link Button External" paragraph appear
  And the "Featured Item URL" field appears
  And the field is required
  And it is a text input
  And the "Button Text" field appears
  And the field will be required
  And the field will be a text input
  And the "Button Alt Text" field appears
  And the field will be optional
  And the field will be a text input
When the user adds a link to the "Primary Call to Action Link" field
Then the user cannot add any more links
 ```

```gherkin
Scenario: Content editor is able to add an "NCIDS Call to Action Strip" to the "Secondary Calls to Action Links" field on the "NCIDS Hero"
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user has added a "NCIDS Hero" to the "Contents" paragraph
When the user adds an "NCIDS Call to Action Strip" to the "Secondary Calls to Action" paragraph
Then the fields for an "NCIDS Call to Action Strip" appear
  And the "Links" field appears
  And it is an existing paragraph
  And it has the options "Add NCIDS Link Button Internal" and "Add NCIDS Link Button External"
  And it defaults to "Add NCIDS Link Button Internal"

Scenario: Content editor is able to add an "NCIDS Link Button" to the "NCIDS Call to Action Strip" in the "Secondary Calls to Action Links" field on the "NCIDS Hero"
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user has added an "NCIDS Hero" to the "Contents" paragraph
  And the user has added an "NCIDS Call to Action Strip" to the "Secondary Calls to Action" paragraph
When the user selects "NCIDS Link Button Internal" for the "Links" paragraph
Then the fields for an "NCIDS Link Button Internal" paragraph appear
  And the "Featured Item" field appears
  And the Featured Item field will be required
  And the Featured Item field will be an entity reference to a node
  And the "Button Text" field appears
  And the field will be required
  And the field will be a text input
  And the "Button Alt Text" field appears
  And the field will be optional
  And the field will be a text input
When the user selects "NCIDS Link Button External" for the "Links" paragraph
Then the fields for an "NCIDS Link Button External" paragraph appear
  And the "Featured Item URL" field appears
  And the field is required
  And it is a text input
  And the "Button Text" field appears
  And the field will be required
  And the field will be a text input
  And the "Button Alt Text" field appears
  And the field will be optional
  And the field will be a text input
When the user adds three links to the "Links" field
Then the user cannot add any more links
```

 ```gherkin
Scenario: Content editor adds a "NCIDS Hero" to a Home and Landing page with fields to display the "NCIDS Hero with Tagline" variant
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to "Contents"
  And the user selects the "Add NCIDS Hero" option
Then the new "NCIDS Hero" paragraph appears
When the user sets the "Theme" field to Light
  And the user sets the "Image" field
  And the user sets the "Tagline" field
Then the user can save the content item
When the user views the Home & Landing Page on cancer.gov
Then the "NCIDS Hero with Tagline" variant of "NCIDS Hero" displays
  And the image displays
  And the Primary Call to Action section displays centered on the image and in the light theme
  And the tagline displays
  And the primary call to action button does not display
  And the supporting call to action buttons do not display
```

 ```gherkin
Scenario: Content editor adds a "NCIDS Hero" to a Home and Landing page with fields to display the "NCIDS Hero with Primary Call to Action" variant
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to "Contents"
  And the user selects the "Add NCIDS Hero" option
Then the new "NCIDS Hero" paragraph appears
When the user sets the "Theme" field to Light
  And the user sets the "Image" field
  And the user sets the "Tagline" field
  And the user adds a link to the "Primary Call to Action Link" field
Then the user can save the content item
When the user views the Home & Landing Page on cancer.gov
Then the "NCIDS Hero with Primary Call to Action - Light" variant of "NCIDS Hero" displays
  And the image displays
  And the Primary Call to Action section displays centered on the image and in the light theme
  And the tagline displays
  And the primary call to action button displays
  And the supporting call to action buttons do not display
```

 ```gherkin
Scenario: Content editor adds a "NCIDS Hero" to a Home and Landing page with fields to display the "NCIDS Hero with Primary and Supporting Calls to Action" variant
Given the user is logged in as an editor
  And the user edits a Home and Landing content item
  And the user expands the dropdown to add a paragraph to "Contents"
  And the user selects the "Add NCIDS Hero" option
Then the new "NCIDS Hero" paragraph appears
When the user sets the "Image" field
  And the user sets the "Tagline" field
  And the user adds a link to the "Primary Call to Action Link" field
  And the user adds three links to the "Secondary Calls to Action Links"
Then the user can save the content item
When the user views the Home & Landing Page on cancer.gov
Then the "NCIDS Hero with Primary and Supporting Calls to Action" variant of "NCIDS Hero" displays
  And the image displays
  And the Primary Call to Action section displays centered on the image and in the light theme
  And the tagline displays
  And the primary call to action button displays
  And the secondary calls to action buttons display
```

### Resources:
Component variant designs
- https://github.com/NCIOCPL/ncids/issues/596
- https://github.com/NCIOCPL/ncids/issues/535
- https://github.com/NCIOCPL/ncids/issues/547

### Notes

---

### Prerequisites
- [x] #3487 

### Tasks
- [ ] Create NCIDS Hero paragraph with fields
- [ ] Create image styles for Hero images (controls maximum/minimum width and height)
- [ ] Twig templates

### Technical Notes
Content modeling:
- [NCIDS Hero](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-hero)

### LINKED ISSUE #3834: NCIDS With Title page style needs optional Subheading field
### Issue description
- Only displays when the NCIDS With Title page style is chosen
  - **NOTE**: This is something that maybe will be done as a spike later.
- Drupal form display order: after Page Title and before Browser Title 
- 255 characters
- Tooltip text: Displays only on Home & Landing Pages using NCIDS with Title style. Displays below H1 Page Title. Used to provide additional context, such as organizational hierarchy or high-level overview for landing pages not using hero with tagline.
- Font should be bold

> **ESTIMATE** TBD

### What's the expected change?
-

### What's the current functionality?
-

### What's the updated acceptance criteria?
```gherkin
```

### Additional details / screenshot
- Screenshot

### Related Tickets
* Original Story Ticket #9999
* Original Story Ticket #9999

### LINKED ISSUE #3864: Home and landing page background should be grey
## Update 5/31/2023
In reviewing our section approach, this ticket is being changed to make it so that the background will not be gray, but instead we will fix all the "gray background" components to use `usa-section--light`. This was because the title for the page needs to be within the `<main>` element when using the NCIDS Landing with Title page style, and for it to be in the `<main>` it needs to be wrapped with a `usa-section`. The title section needs to be white, so if the background is white, then the default `usa-section` will use that background color.

### Issue description
The background color for home and landing pages is currently displaying as white. ~This should be updated to be the `base-lightest (#1b1b1b)` theme token. Additionally, the component rows should be updated to use `usa-section` without any light color settings (no `usa-section--light`).~ Based on the update, we need to apply `usa-section--light` to promo blocks and other elements that are not using the gray.

> **ESTIMATE** TBD

### Steps to reproduce the issue

1.  Go to https://ncigovcdode556.prod.acquia-sites.com/ncids-guide-card-test

### What's the expected result?
- All components in the body content should have the `base-lightest` background.

### What's the actual result?
- The promo block components have a white background.

### Additional details / screenshot
Screenshot 2023-05-17 at 10 48 18 AM

### Related Tickets
* Issue #9999
* Issue #9999

### LINKED ISSUE #3866: Landing pages a11y issues
- [ ] Main content areas should always use `<main>` so it has the correct role. 
    ```diff
    - <div class="main-content type {{ page_type }}" id="main-content">...</div>
    + <main class="main-content type {{ page_type }}" id="main-content">...</main>
    
- [ ] Sections should always have a `labelledby={{ some id }}` attribute so we always know a section is used correctly. If there is no label or header, `div` should be used.
    ```diff
    - <section class="usa-section">...</section>
    + <section class="usa-section" labelledby="featureCardRow1">...</section>
    ```
    

Should go ahead and make sure other templates are also correct, but this ticket targets release 2 components, e.g. promo blocks, feature cards, guide cards.

### PR DESCRIPTION
**Pull request by dev-rana-publicis**
ODE: https://ncigovcdode559.prod.acquia-sites.com/

Closes #3866
Closes #3864
Closes #3834
Closes #3487
Closes #3488

### CODE REVIEW

**Review by bennettcc — APPROVED** [review]

**Comment by bennettcc** [comment]
Approving this on behalf of product and IA, Fed front end testers - passed on the ODE

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-internal--link-button-cta.html.twig [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  {% set buttonText = paragraph.field_button_text.value %}
+  {% if buttonText is empty %}
+    {% set buttonText = featured_item_node.field_card_title[0].value %}
+    {% if buttonText is empty %}
+      {% set buttonText = featured_item_node.field_browser_title[0].value %}
+    {% endif %}
+  {% endif %}
+
+  {% set url = path('entity.node.canonical', {'node':  content.field_internal_link[0]["#node"].nid[0].value}) %}
+  <a href="{{ url }}" alt="{{ paragraph.field_button_alt.value }}" class="usa-button">
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-internal--link-button-cta.html.twig:17]: See the external link item comment for the `alt` attribute issue.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-hero.html.twig [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <div class="nci-hero__cta {{ buttonClassName }}">
+        <h2 class="nci-hero__cta-tagline" id="{{ id }}">{{ paragraph.field_tagline.value }}</h2>
+        {{ content.field_link_button }}
+      </div>
+    </div>
+  </div>
+  {% if content.field_cta_link_buttons[0] %}
+    <div class="usa-section usa-section--dark usa-section--nci-cta-strip cgdp-cta-strip">
+      <div class="grid-container">
+        <ul class="nci-cta-strip" aria-label="Call to action">
```
Comment (dev-rana-publicis) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-hero.html.twig:48]: refer back to Bryan's comment

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-hero.html.twig [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    </picture>
+    <div class="nci-hero__cta-container">
+      <div class="nci-hero__cta {{ buttonClassName }}">
+        <h2 class="nci-hero__cta-tagline" id="{{ id }}">{{ paragraph.field_tagline.value }}</h2>
+        {{ content.field_link_button }}
+      </div>
+    </div>
+  </div>
+  {% if content.field_cta_link_buttons[0] %}
+    <div class="usa-section usa-section--dark usa-section--nci-cta-strip cgdp-cta-strip">
```
Comment (dev-rana-publicis) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-hero.html.twig:46]: remove `cgdp-cta-strip`

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-external--link-button-hero.html.twig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<a href="{{ paragraph.field_external_link.uri }}" alt="{{ paragraph.field_button_alt.value }}" class="nci-hero__cta-button usa-button">
```
Comment (dev-rana-publicis) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-external--link-button-hero.html.twig:1]: refer back to Bryan's comments

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Changes so far...

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-internal--link-button-hero.html.twig [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  {% if buttonText is empty %}
+    {% set buttonText = featured_item_node.field_card_title[0].value %}
+    {% if buttonText is empty %}
+      {% set buttonText = featured_item_node.field_browser_title[0].value %}
+    {% endif %}
+  {% endif %}
+
+  {% set url = path('entity.node.canonical', {'node':  featured_item_node.nid[0].value}) %}
+
+  <a href="{{ url }}" alt="{{ paragraph.field_button_alt.value }}" class="nci-hero__cta-button usa-button">
```
Comment (dev-rana-publicis) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-internal--link-button-hero.html.twig:18]: Bryan comment

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-external--link-button-cta.html.twig [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<a href="{{ paragraph.field_external_link.uri }}" alt="{{ paragraph.field_button_alt.value }}" class="usa-button">
```
[the remark this answers is not in the document]
Comment (dev-rana-publicis) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-link-button-external--link-button-cta.html.twig:1]: Drupal just renders out whatever is set on the field. The link button nor the cta strip breaks if there is a single or double quote in the aria-label.

**Comment by dev-rana-publicis** [comment]
Non Hero and CTA Examples:
1. NCIDS with title: http://www.devbox/ncids-landing-with-title
    - title (0px top, 32px bot)
    - r2.1 component double spacing (16px)
2. NCIDS without title: http://www.devbox/ncids-guide-card-test
    - r2.1 component (32px)
3. NCIDS with title: {{ add on to http://www.devbox/ncids-landing-with-title}}
    - title (0px top, 32px bot)
    - r2.1 component (16px)
    - r2.1 component (16px)
4. NCIDS without title: http://www.devbox/ncids-external-feature-card-test
    - r2.1 component (32px)
    - r2.1 component (16px)

HERO EXAMPLES:
1. NCIDS with title: http://www.devbox/ncids-hero-default-test
    - title (0px top, 32px bot)
    - hero (0px top, 0px bot)
    - r2.1 component double spacing (32px)
2. NCIDS without title: http://www.devbox/ncids-hero-cta-test
    - hero (0px top, 0px bot)
    - r2.1 component (32px)
3. NCIDS with title: {{ need to make }}
    - title (0px top, 32px bot)
    - hero (0px top, 0px bot)
    - r2.1 component (32px)
    - r2.1 component (16px)
4. NCIDS without title: http://www.devbox/ncids-hero-grants-training-test
    - hero (0px top, 0px bot)
    - r2.1 component (32px)
    - r2.1 component (16px)

CTA EXAMPLES:
1. NCIDS with title: http://www.devbox/ncids-hero-default-test
    - title (0px top, 32px bot)
    - CTA (16px top, 16px bot)
    - r2.1 component double spacing (16px)
2. NCIDS without title: http://www.devbox/ncids-hero-cta-test
    - CTA (32px top, 16px bot)
    - r2.1 component (16px)
3. NCIDS with title: {{ need to make }}
    - title (0px top, 32px bot)
    - CTA (16px top, 16px bot)
    - r2.1 component (16px)
    - r2.1 component (16px)
4. NCIDS without title: http://www.devbox/ncids-hero-grants-training-test
    - CTA (32px top, 16px bot)
    - r2.1 component (16px)
    - r2.1 component (16px)

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**